src/uct/sm/base/sm_iface.c

```diff
@@ -151,12 +151,13 @@ UCS_CLASS_INIT_FUNC(uct_sm_iface_t, uct_iface_ops_t *ops, uct_md_h md,
         return UCS_ERR_UNSUPPORTED;
     }
 
-    UCS_CLASS_CALL_SUPER_INIT(uct_base_iface_t, ops, md, worker, params,
-                              tl_config
-                              UCS_STATS_ARG((params->field_mask &
-                                             UCT_IFACE_PARAM_FIELD_STATS_ROOT) ?
-                                            params->stats_root : NULL)
-                              UCS_STATS_ARG(params->mode.device.dev_name));
+    UCS_CLASS_CALL_SUPER_INIT(
+            uct_base_iface_t, ops, &uct_iface_default_internal_ops, md, worker,
+            params,
+            tl_config UCS_STATS_ARG(
+                    (params->field_mask & UCT_IFACE_PARAM_FIELD_STATS_ROOT) ?
+                            params->stats_root :
```

Comment:
maybe remove stats_root argument from base_iface constructor, and let it take the stats root from params structure

Reply:
Sounds good. Maybe in a separate PR?

Reply:
yep